matrix-sdk/src/main/java/org/matrix/androidsdk/data/timeline/TimelineEventListeners.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.androidsdk.data.timeline;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.matrix.androidsdk.data.RoomState;
+import org.matrix.androidsdk.rest.model.Event;
+import org.matrix.androidsdk.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Handle the timeline event listeners
+ * Is responsible for dispatching events
+ */
+class TimelineEventListeners {
+
+    private static final String LOG_TAG = TimelineEventListeners.class.getSimpleName();
+
+    // The inner listeners
+    private final List<EventTimeline.Listener> mListeners = new ArrayList<>();
+
+    /**
+     * Add an events listener.
+     *
+     * @param listener the listener to add.
+     */
+    public void add(@Nullable final EventTimeline.Listener listener) {
+        if (listener != null) {
+            synchronized (this) {
+                if (mListeners.indexOf(listener) == -1) {
```

Comment:
Use `contains()`

Reply:
Yeah, probably need some rework, just copy/paste at the moment

Reply:
done